cmake/FindSDL2.cmake

```diff
@@ -65,71 +65,88 @@
 # (To distribute this file outside of CMake, substitute the full
 #  License text for the above reference.)
 
-if(CMAKE_SIZEOF_VOID_P EQUAL 8)
-  set(SDL2_ARCH_64 TRUE)
-  set(SDL2_PROCESSOR_ARCH "x64")
-else()
-  set(SDL2_ARCH_64 FALSE)
-  set(SDL2_PROCESSOR_ARCH "x86")
-endif(CMAKE_SIZEOF_VOID_P EQUAL 8)
-
 GET_SDL_EXT_DIR(SDL_EXT_DIR "")
 
-if(MINGW AND DEFINED SDL_EXT_DIR)
-    if(SDL2_ARCH_64)
-	  set(SDL_MINGW_EXT_DIR "${SDL_EXT_DIR}/x86_64-w64-mingw32")
+IF(ANDROID_BUILD)
+    STRING(TOLOWER ${CMAKE_BUILD_TYPE} ANDROID_BUILD_DIR)
+    SET(SDL2_LIBRARY SDL2)
+    SET(SDL2_ANDROID_HOOK ${SDL_EXT_DIR}/src/main/android/SDL_android_main.c)
+    ADD_LIBRARY(${SDL2_LIBRARY} SHARED IMPORTED)
+    SET_TARGET_PROPERTIES(${SDL2_LIBRARY} PROPERTIES IMPORTED_LOCATION
+    ${PROJECT_SOURCE_DIR}/android/distribution/android/SDL2/intermediates/ndkBuild/${ANDROID_BUILD_DIR}/obj/local/${ANDROID_ABI}/libSDL2.so)
```

Comment:
It's the path where Gradle stores the built files, and `FIND_FILE` has never really worked that well on CMake for Android...

Reply:
After having a look at this, I don't really know of a better way do integrate our existing code with Gradle. So this will have to stick :\

Reply:
It is weird to me too, but I also failed to find a way around it.  Strange Android build magic because it is merging native and Java code.  I think there is supposed to be a way it can be improved using mklink though to connect the SDL library to the julius android path.

Reply:
Yeah, Android doesn't have much support for native libraries. It would also be great if we could store the pre built SDL libraries somewhere and reuse them, as it would drastically reduce the build times for travis.